File: daily/Linked List/1669-Merge-in-between-linked-lists.py
class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        dummy = ListNode(0, list1)
        before = dummy

        for _ in range(a):
            before = before.next
        
        curr = before.next
        for _ in range(b - a + 1):
            curr = curr.next
        
        list2_curr = list2
        while list2_curr.next:
            list2_curr = list2_curr.next
            
        list2_curr.next = curr
        before.next = list2
        return dummy.next

File: daily/Linked List/1669-Merge-in-between-linked-lists.go
package daily

func mergeInBetween(list1 *ListNode, a int, b int, list2 *ListNode) *ListNode {
	dummy := &ListNode{0, list1}
	before := dummy

	for i := 0; i < a; i++ {
		before = before.Next
	}

	curr := before.Next
	for i := 0; i < b-a+1; i++ {
		curr = curr.Next
	}

	list2Curr := list2
	for list2Curr.Next != nil {
		list2Curr = list2Curr.Next
	}

	list2Curr.Next = curr
	before.Next = list2
	return dummy.Next
}
